robot_hardware:
  - panda_left
  - panda_right
  - panda_top

panda_left:
  type: franka_hw/FrankaCombinableHW
  arm_id: panda_left
  joint_names:
    - panda_left_joint1
    - panda_left_joint2
    - panda_left_joint3
    - panda_left_joint4
    - panda_left_joint5
    - panda_left_joint6
    - panda_left_joint7
  # Configure the threshold angle for printing joint limit warnings.
  joint_limit_warning_threshold: 0.1 # [rad]
  # Activate rate limiter? [true|false]
  rate_limiting: true
  # Cutoff frequency of the low-pass filter. Set to >= 1000 to deactivate.
  cutoff_frequency: 1000
  # Internal controller for motion generators [joint_impedance|cartesian_impedance]
  internal_controller: joint_impedance
  # Used to decide whether to enforce realtime mode [enforce|ignore]
  realtime_config: ignore
  # Configure the initial defaults for the collision behavior reflexes.
  collision_config:
    lower_torque_thresholds_acceleration: [20.0, 20.0, 18.0, 18.0, 16.0, 14.0, 12.0]  # [Nm]
    upper_torque_thresholds_acceleration: [20.0, 20.0, 18.0, 18.0, 16.0, 14.0, 12.0]  # [Nm]
    lower_torque_thresholds_nominal: [20.0, 20.0, 18.0, 18.0, 16.0, 14.0, 12.0]  # [Nm]
    upper_torque_thresholds_nominal: [20.0, 20.0, 18.0, 18.0, 16.0, 14.0, 12.0]  # [Nm]
    lower_force_thresholds_acceleration: [45.0, 45.0, 45.0, 35.0, 35.0, 35.0]  # [N, N, N, Nm, Nm, Nm]
    upper_force_thresholds_acceleration: [45.0, 45.0, 45.0, 35.0, 35.0, 35.0]  # [N, N, N, Nm, Nm, Nm]
    lower_force_thresholds_nominal: [45.0, 45.0, 45.0, 35.0, 35.0, 35.0]  # [N, N, N, Nm, Nm, Nm]
    upper_force_thresholds_nominal: [45.0, 45.0, 45.0, 35.0, 35.0, 35.0]  # [N, N, N, Nm, Nm, Nm]


panda_right:
  type: franka_hw/FrankaCombinableHW
  arm_id: panda_right
  joint_names:
    - panda_right_joint1
    - panda_right_joint2
    - panda_right_joint3
    - panda_right_joint4
    - panda_right_joint5
    - panda_right_joint6
    - panda_right_joint7
  # Configure the threshold angle for printing joint limit warnings.
  joint_limit_warning_threshold: 0.1 # [rad]
  # Activate rate limiter? [true|false]
  rate_limiting: true
  # Cutoff frequency of the low-pass filter. Set to >= 1000 to deactivate.
  cutoff_frequency: 1000
  # Internal controller for motion generators [joint_impedance|cartesian_impedance]
  internal_controller: joint_impedance
  # Used to decide whether to enforce realtime mode [enforce|ignore]
  realtime_config: ignore
  # Configure the initial defaults for the collision behavior reflexes.
  collision_config:
    # lower_torque_thresholds_acceleration: [100.0, 100.0, 100.0, 100.0, 100.0, 100.0, 100.0]  # [Nm]
    # upper_torque_thresholds_acceleration: [100.0, 100.0, 100.0, 100.0, 100.0, 100.0, 100.0]  # [Nm]
    # lower_torque_thresholds_nominal: [100.0, 100.0, 100.0, 100.0, 100.0, 100.0, 100.0]  # [Nm]
    # upper_torque_thresholds_nominal: [100.0, 100.0, 100.0, 100.0, 100.0, 100.0, 100.0]  # [Nm]
    # lower_torque_thresholds_acceleration: [20.0, 20.0, 18.0, 18.0, 16.0, 14.0, 12.0]  # [Nm]
    # upper_torque_thresholds_acceleration: [20.0, 20.0, 18.0, 18.0, 16.0, 14.0, 12.0]  # [Nm]
    # lower_torque_thresholds_nominal: [20.0, 20.0, 18.0, 18.0, 16.0, 14.0, 12.0]  # [Nm]
    # upper_torque_thresholds_nominal: [20.0, 20.0, 18.0, 18.0, 16.0, 14.0, 12.0]  # [Nm]
    lower_torque_thresholds_acceleration: [40.0, 40.0, 40.0, 40.0, 30.0, 30.0, 30.0]  # [Nm]
    upper_torque_thresholds_acceleration: [40.0, 40.0, 40.0, 40.0, 30.0, 30.0, 30.0]  # [Nm]
    lower_torque_thresholds_nominal: [40.0, 40.0, 40.0, 40.0, 30.0, 30.0, 30.0]  # [Nm]
    upper_torque_thresholds_nominal: [40.0, 40.0, 40.0, 40.0, 30.0, 30.0, 30.0]  # [Nm] 
    # lower_force_thresholds_acceleration: [100.0, 100.0, 100.0, 100.0, 100.0, 100.0]  # [N, N, N, Nm, Nm, Nm]
    # upper_force_thresholds_acceleration: [100.0, 100.0, 100.0, 100.0, 100.0, 100.0]  # [N, N, N, Nm, Nm, Nm]
    # lower_force_thresholds_nominal: [100.0, 100.0, 100.0, 100.0, 100.0, 100.0]  # [N, N, N, Nm, Nm, Nm]
    # upper_force_thresholds_nominal: [100.0, 100.0, 100.0, 100.0, 100.0, 100.0]  # [N, N, N, Nm, Nm, Nm]
    lower_force_thresholds_acceleration: [45.0, 45.0, 45.0, 35.0, 35.0, 35.0]  # [N, N, N, Nm, Nm, Nm]
    upper_force_thresholds_acceleration: [45.0, 45.0, 45.0, 35.0, 35.0, 35.0]  # [N, N, N, Nm, Nm, Nm]
    lower_force_thresholds_nominal: [45.0, 45.0, 45.0, 35.0, 35.0, 35.0]  # [N, N, N, Nm, Nm, Nm]
    upper_force_thresholds_nominal: [45.0, 45.0, 45.0, 35.0, 35.0, 35.0]  # [N, N, N, Nm, Nm, Nm]

panda_top:
  type: franka_hw/FrankaCombinableHW
  arm_id: panda_top
  joint_names:
    - panda_top_joint1
    - panda_top_joint2
    - panda_top_joint3
    - panda_top_joint4
    - panda_top_joint5
    - panda_top_joint6
    - panda_top_joint7
  # Configure the threshold angle for printing joint limit warnings.
  joint_limit_warning_threshold: 0.1 # [rad]
  # Activate rate limiter? [true|false]
  rate_limiting: true
  # Cutoff frequency of the low-pass filter. Set to >= 1000 to deactivate.
  cutoff_frequency: 1000
  # Internal controller for motion generators [joint_impedance|cartesian_impedance]
  internal_controller: joint_impedance
  # Used to decide whether to enforce realtime mode [enforce|ignore]
  realtime_config: ignore
  # Configure the initial defaults for the collision behavior reflexes.
  collision_config:
  #   lower_torque_thresholds_acceleration: [20.0, 20.0, 18.0, 18.0, 16.0, 14.0, 12.0]  # [Nm]
  #   upper_torque_thresholds_acceleration: [20.0, 20.0, 18.0, 18.0, 16.0, 14.0, 12.0]  # [Nm]
  #   lower_torque_thresholds_nominal: [20.0, 20.0, 18.0, 18.0, 16.0, 14.0, 12.0]  # [Nm]
  #   upper_torque_thresholds_nominal: [20.0, 20.0, 18.0, 18.0, 16.0, 14.0, 12.0]  # [Nm]
  #   lower_force_thresholds_acceleration: [45.0, 45.0, 45.0, 35.0, 35.0, 35.0]  # [N, N, N, Nm, Nm, Nm]
  #   upper_force_thresholds_acceleration: [45.0, 45.0, 45.0, 35.0, 35.0, 35.0]  # [N, N, N, Nm, Nm, Nm]
  #   lower_force_thresholds_nominal: [45.0, 45.0, 45.0, 35.0, 35.0, 35.0]  # [N, N, N, Nm, Nm, Nm]
  #   upper_force_thresholds_nominal: [45.0, 45.0, 45.0, 35.0, 35.0, 35.0]  # [N, N, N, Nm, Nm, Nm]
  # For short bolt
    lower_torque_thresholds_acceleration: [40.0, 40.0, 40.0, 40.0, 30.0, 30.0, 30.0]  # [Nm]
    upper_torque_thresholds_acceleration: [40.0, 40.0, 40.0, 40.0, 30.0, 30.0, 30.0]  # [Nm]
    lower_torque_thresholds_nominal: [40.0, 40.0, 40.0, 40.0, 30.0, 30.0, 30.0]  # [Nm]
    upper_torque_thresholds_nominal: [40.0, 40.0, 40.0, 40.0, 30.0, 30.0, 30.0]  # [Nm]
    lower_force_thresholds_acceleration: [60.0, 60.0, 60.0, 60.0, 60.0, 60.0]  # [N, N, N, Nm, Nm, Nm]
    upper_force_thresholds_acceleration: [60.0, 60.0, 60.0, 60.0, 60.0, 60.0]  # [N, N, N, Nm, Nm, Nm]
    lower_force_thresholds_nominal: [60.0, 60.0, 60.0, 60.0, 60.0, 60.0]  # [N, N, N, Nm, Nm, Nm]
    upper_force_thresholds_nominal: [60.0, 60.0, 60.0, 60.0, 60.0, 60.0]  # [N, N, N, Nm, Nm, Nm]